--- amarok-1.1.1.orig/configure.in	2004-10-09 14:52:14.000000000 +0200
+++ amarok-1.1.1/configure.in	2004-11-01 23:05:13.732752488 +0100
@@ -441,7 +441,7 @@
 
 if test x$included_sqlite = xno; then
     if test x$PKGCONFIGFOUND = xyes; then
-        KDE_PKG_CHECK_MODULES(SQLITE, sqlite >= 2.8, have_sqlite=yes,have_sqlite=no)
+        KDE_PKG_CHECK_MODULES(SQLITE, sqlite3 >= 3.0, have_sqlite=yes,have_sqlite=no)
 
         if test x$have_sqlite = xyes; then
             ## AC_DEFINE(HAVE_SQLITE, 1, [have SQLite database library])
--- amarok-1.1.1.orig/amarok/configure.in.in	2004-10-09 14:52:09.000000000 +0200
+++ amarok-1.1.1/amarok/configure.in.in	2004-11-01 23:50:27.301227376 +0100
@@ -368,7 +368,7 @@
 
 if test x$included_sqlite = xno; then
     if test x$PKGCONFIGFOUND = xyes; then
-        KDE_PKG_CHECK_MODULES(SQLITE, sqlite >= 2.8, have_sqlite=yes,have_sqlite=no)
+        KDE_PKG_CHECK_MODULES(SQLITE, sqlite3 >= 3.0, have_sqlite=yes,have_sqlite=no)
 
         if test x$have_sqlite = xyes; then
             ## AC_DEFINE(HAVE_SQLITE, 1, [have SQLite database library])
