diff -ur amarok-1.4.1-beta1/amarok/src/mediadevice/njb/njbmediadevice.h amarok-1.4.1-beta1.new/amarok/src/mediadevice/njb/njbmediadevice.h
--- amarok-1.4.1-beta1/amarok/src/mediadevice/njb/njbmediadevice.h	2006-06-18 20:56:09.000000000 +0000
+++ amarok-1.4.1-beta1.new/amarok/src/mediadevice/njb/njbmediadevice.h	2006-06-20 10:08:31.000000000 +0000
@@ -17,7 +17,7 @@
 // #include <mediadevice.h>
 #include <mediabrowser.h>
 #include <transferdialog.h>
-#include <libnjb.h>
+#include <libnjb/libnjb.h>
 
 #include <qptrlist.h>
 #include <qlistview.h>
diff -ur amarok-1.4.1-beta1/amarok/src/mediadevice/njb/playlist.h amarok-1.4.1-beta1.new/amarok/src/mediadevice/njb/playlist.h
--- amarok-1.4.1-beta1/amarok/src/mediadevice/njb/playlist.h	2006-06-18 20:56:09.000000000 +0000
+++ amarok-1.4.1-beta1.new/amarok/src/mediadevice/njb/playlist.h	2006-06-20 10:28:27.000000000 +0000
@@ -20,7 +20,7 @@
 #define __playlist_h__
 
 // libnjb
-#include <libnjb.h>
+#include <libnjb/libnjb.h>
 
 // Qt
 #include <qstringlist.h>
diff -ur amarok-1.4.1-beta1/amarok/src/mediadevice/njb/track.h amarok-1.4.1-beta1.new/amarok/src/mediadevice/njb/track.h
--- amarok-1.4.1-beta1/amarok/src/mediadevice/njb/track.h	2006-06-18 20:56:09.000000000 +0000
+++ amarok-1.4.1-beta1.new/amarok/src/mediadevice/njb/track.h	2006-06-20 10:28:44.000000000 +0000
@@ -24,7 +24,7 @@
 #include <qstring.h>
 
 // libnjb
-#include <libnjb.h>
+#include <libnjb/libnjb.h>
 
 #include "metabundle.h"
 
