--- src/context/engines/wikipedia/WikipediaEngine.h.orig	2009-02-10 19:58:15.000000000 +0100
+++ src/context/engines/wikipedia/WikipediaEngine.h	2009-02-16 14:17:57.968771852 +0100
@@ -21,6 +21,7 @@
 #include <context/DataEngine.h>
 
 #include <KIO/Job>
+#include <QLocale>
 
 /**
     This class provide Wikipedia data for use in Context applets. 
@@ -80,7 +81,7 @@
     QString m_wikiCurrentEntry;
     QString m_wikiCurrentUrl;
     QString m_wikiLanguages;
-    QString m_wikiLocale;
+    QLocale m_wikiLang;
     // stores what features are enabled
     bool m_requested;
     QStringList m_sources;
--- src/context/engines/wikipedia/WikipediaEngine.cpp.orig	2009-02-10 19:58:15.000000000 +0100
+++ src/context/engines/wikipedia/WikipediaEngine.cpp	2009-02-16 14:19:52.111853566 +0100
@@ -27,7 +27,6 @@
     , ContextObserver( ContextView::self() )
     , m_wikiJob( 0 )
     , m_currentSelection( "artist" )
-    , m_wikiLocale( "en" )
     , m_requested( true )
     , m_sources( "current" )
     , m_triedRefinedSearch( false )
@@ -329,7 +328,7 @@
 inline QString
 WikipediaEngine::wikiLocale() const
 {
-    return !m_wikiLocale.isEmpty() ? m_wikiLocale : QString( "en" );
+	return m_wikiLang.name().split('_')[0];
 }
 
 inline QString
@@ -339,6 +338,8 @@
         return " (band)";
     else if( wikiLocale() == "de" )
         return " (Band)";
+    else if( wikiLocale() == "pl" )
+        return " (Kapela)";
     return QString();
 }
 
