--- amarok-1.4.7/amarok/src/metadata/mp4/mp4file.cpp.orig	2007-08-07 19:48:28.000000000 +0200
+++ amarok-1.4.7/amarok/src/metadata/mp4/mp4file.cpp	2007-12-05 08:55:17.346943004 +0100
@@ -156,11 +156,7 @@
     }
 #endif
 
-    if(!MP4Close(handle))
-    {
-        fprintf(stderr, "close failed\n");
-    }
-
+    MP4Close(handle);
     mp4file = MP4Read(name());
     if(mp4file == MP4_INVALID_FILE_HANDLE)
     {
